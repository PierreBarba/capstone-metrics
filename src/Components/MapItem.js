import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const MapItem = (props) => {
  const { name, flag, population } = props;

  return (
    <Link to={`Specification/${name}`} className="text">
      <div className="card" id="flex">
        <img className="flags" src={flag} alt="flag" />
        <div className="info">
          <h2 className="card-name">{name}</h2>
          <p className="card-p">
            Population:
            {' '}
            {population}
            {' '}
            habs
          </p>
        </div>
      </div>
    </Link>
  );
};

export default MapItem;

MapItem.propTypes = {
  name: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
  flag: PropTypes.string.isRequired,
};
