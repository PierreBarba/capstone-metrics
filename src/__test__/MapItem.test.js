import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import MapItem from '../Components/MapItem';
import '@testing-library/jest-dom';
import store from '../Redux/ConfigureStore';

const Country = {
  name: 'Ukraine',
  population: 44134693,
  flag: 'https://flagcdn.com/is.svg',
  id: 'e115687c-660f-48e8-8da0-bf9809a9140c',
  subregion: 'Eastern Europe',
  coordenates: {
    lat: 49,
    long: 32,
  },
  soweek: 'monday',
  capital: 'Kyiv',
  coarms: 'https://mainfacts.com/media/images/coats_of_arms/ua.svg',
};

describe('Test page deployment', () => {
  test('MapItem components matches snapshot', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <MapItem
            name={Country.name}
            population={Country.population}
            flag={Country.flag}
          />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
