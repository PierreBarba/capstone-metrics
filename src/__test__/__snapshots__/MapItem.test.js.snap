// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test page deployment MapItem components matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="text"
        href="/Specification/Ukraine"
      >
        <div
          class="card"
          id="flex"
        >
          <img
            alt="flag"
            class="flags"
            src="https://flagcdn.com/is.svg"
          />
          <div
            class="info"
          >
            <h2
              class="card-name"
            >
              Ukraine
            </h2>
            <p
              class="card-p"
            >
              Population:
               
              44134693
               
              habs
            </p>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="text"
      href="/Specification/Ukraine"
    >
      <div
        class="card"
        id="flex"
      >
        <img
          alt="flag"
          class="flags"
          src="https://flagcdn.com/is.svg"
        />
        <div
          class="info"
        >
          <h2
            class="card-name"
          >
            Ukraine
          </h2>
          <p
            class="card-p"
          >
            Population:
             
            44134693
             
            habs
          </p>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
